### DISCUSSION TITLE
Upload component

### DISCUSSION THREAD

**Discussion by ObaPaguei** [rsuite/rsuite#3218] [Q&A]
Hey guys!

I would like to know how do I get the file that is being UPLOAD with the Upload component? I need this to store in memory and then save the complete list.

**Comment by sijirama** [comment]
i believe some props in the Uploader component such as the onUpload prop have the file passed into it, you can set it to a state then use it.

**Comment by sijirama** [comment]
i actually had issues with the action prop, so what i did was make all my logic in the file Uploader onUpload prop since the file is passed as an argument there.
